docs/tutorials/td-tutorial.md

```diff
@@ -0,0 +1,226 @@
+# Send data securely to Arm Treasure Data
+
+There are two ways to send data securely from Mbed OS to Treasure Data:
+
+- HTTPS library - Send data directly to the Treasure Data REST API.
+- fluentd using fluent logger library - Send data to a hosted fluentd instance that aggregates and forwards the data on to your treasure data account.
+
+Both libraries are secured with Arm Mbed TLS in transit and are equally secure. We recommend the HTTPS library for development and the fluentd library for production. The tradeoff between the two is size of code on chip, size of data in transit and setup complexity: 
+
+- Code size on chip - The HTTPS library is ~50KB of ROM space on chip, this due to the HTTP stack. Both libraries use Mbed TLS to secure the connections, which is ~7KB per connection on your stack for both libraries.
+- Data size in transit - The HTTPS library sends data as a ASCII JSON string. The fluend library uses MessagePack (binary encoded json) across a TLS connection. This means that on average the fluentd library will use less bandwidth to send an equivalent message. When you pay per byte transmitted from both your power budget and data plan it matters.
+- Maintenance - Initially, it may be simpler to setup the HTTPS library on a device and have it send data directly to treasure data, but what if you want to change what the device is doing or how its data is reported? If you are using the HTTPS library you will need to issue a firmware update to every device to change how it formats its data, but if you are using a fluend server you can simply modify the fluentd config file on the server to change how data is formatted/processed.
+
+The following steps show how to send data using first the HTTPS library and then using fluentd.
+
+## HTTPS library
+
+To use the HTTPS library, use this program: https://github.com/blackstoneengineering/mbed-os-example-treasuredata-rest. This program turns on Mbed OS device statistics by enabling the `MBED_ALL_STATS_ENABLED` macro and then send heap/CPU/stack/system information to Treasure Data.
+
+https://www.youtube.com/watch?v=_tqD6GLMHQA
+
+### Import code
+
+You can compile the program through any of our three development tools:
+
+- [Arm Mbed Studio](https://os.mbed.com/studio/).
+- Arm Online Compiler - `ide.mbed.com/compiler?import=https://github.com/blackstoneengineering/mbed-os-example-treasuredata-rest`
+- Arm Mbed CLI (offline) - `mbed import https://github.com/blackstoneengineering/mbed-os-example-treasuredata-rest`
+
+### Setup variables
+
+1. Configure the Treasure Data API key in `mbed_app.json` by change the `api-key` variable:
+
+   ```
+   "api-key":{
+   
+   
+   "help": "REST API Key for Treasure Data",
+    
+   "value": "\"REPLACE_WITH_YOUR_KEY\""
+    
+   },
+    
+   ```
+
+1. Wi-Fi credentials (optional): If you're using Wi-Fi, add your SSID/password. If you are using ethernet, you do not need to add Wi-Fi credentials.
```

Comment:
Query: Is this actually optional? It sounds like it's not if you're using Wi-Fi. Also, do you add your SSID *and* password, or your SSID *or* password?

Reply:
Its to imply that this works on Ethernet / Wifi / Cellular / etc. Its optional if you're not using wifi.

Reply:
I would remove (optional); I agree with Amanda that it's confusing.